src/components/Search/SearchRouter/SearchRouter.tsx

```diff
@@ -1,71 +1,125 @@
+import SearchRouterList from './SearchRouterList';
 
 const SEARCH_DEBOUNCE_DELAY = 150;
 
 function SearchRouter() {
     const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const [betas] = useOnyx(`${ONYXKEYS.BETAS}`);
+    const [recentSearches] = useOnyx(ONYXKEYS.RECENT_SEARCHES);
 
     const {isSmallScreenWidth} = useResponsiveLayout();
     const {isSearchRouterDisplayed, closeSearchRouter} = useSearchRouterContext();
+    const listRef = useRef<SelectionListHandle>(null);
 
+    const [textInputValue, setTextInputValue] = useState('');
     const [userSearchQuery, setUserSearchQuery] = useState<SearchQueryJSON | undefined>(undefined);
+    const contextualReportID = useNavigationState<Record<string, {reportID: string}>, string | undefined>((state) => {
+        return state?.routes.at(-1)?.params?.reportID;
+    });
+    const sortedRecentSearches = useMemo(() => {
+        return Object.values(recentSearches ?? {}).sort((a, b) => {
+            const dateA = new Date(a.timestamp);
+            const dateB = new Date(b.timestamp);
+            return dateB.getTime() - dateA.getTime();
+        });
+    }, [recentSearches]);
+
+    const {options, areOptionsInitialized} = useOptionsList({
+        shouldInitialize: true,
+    });
+    const searchOptions = useMemo(() => {
+        if (!areOptionsInitialized) {
+            return {recentReports: [], personalDetails: [], userToInvite: null, currentUserOption: null, categoryOptions: [], tagOptions: [], taxRatesOptions: []};
+        }
+        return OptionsListUtils.getSearchOptions(options, '', betas ?? []);
+    }, [areOptionsInitialized, betas, options]);
+
+    const contextualReportData = contextualReportID ? searchOptions.recentReports?.find((option) => option.reportID === contextualReportID) : undefined;
 
     const clearUserQuery = () => {
+        setTextInputValue('');
         setUserSearchQuery(undefined);
     };
 
-    const onSearchChange = debounce((userQuery: string) => {
-        if (!userQuery) {
-            clearUserQuery();
-            return;
-        }
-
-        const queryJSON = SearchUtils.buildSearchQueryJSON(userQuery);
-
-        if (queryJSON) {
-            // eslint-disable-next-line
-            console.log('parsedQuery', queryJSON);
-
-            setUserSearchQuery(queryJSON);
-        } else {
-            // Handle query parsing error
-        }
-    }, SEARCH_DEBOUNCE_DELAY);
+    const onSearchChange = useMemo(
+        // eslint-disable-next-line react-compiler/react-compiler
+        () =>
+            debounce((userQuery: string) => {
+                if (!userQuery) {
+                    clearUserQuery();
+                    listRef.current?.updateAndScrollToFocusedIndex(-1);
+                    return;
+                }
+                listRef.current?.updateAndScrollToFocusedIndex(0);
+                const queryJSON = SearchUtils.buildSearchQueryJSON(userQuery);
```

Comment:
It was initially implemented like this by Mateusz. Right now it may not feel like the best idea, but when we will add autocomplete we will have to parse onChange to provide autocomplete suggestions. 
I can refactor it to not be parsed on every change but we will have to roll it back when we implement 2.5, so should I do it then? WDYT @luacmartins

Reply:
Yea, I think we should refactor it right now. I just started working on the autocomplete doc and I think we might want a different parser for it that only captures the filter key and checks for matches, without building the AST

Reply:
I already started refactoring it, but now I have second thoughts.
Parser with or without AST will have to parse input on text change to provide suggestions, so after deleting this we will have to add this once more. Also this code is already working on main and is not that heavy performance-wise.
I could delete Log.alert and console.logs so we don't log things on every keystroke and don't clutter the terminal, but I'm highly in favour of leaving this code as it is.

Reply:
hey it was me that originally introduce the code that parses after every keypress.
Also please note that there is 150ms debounce on this function, so that we do not run it  _too_ often.

In general I was feeling that whatever user types in should go through parser as it feels safer and gives us more information about the query.
Yes its true that right now we have no autocomplete of any kind yet, but it's just easier to work on parsed object.
If you worry about speed and computation - I wouldn't at this point, the parser is very fast and it will run only several times.

Also like @SzymczakJ said - in the long there will be **a** parser in this place, so that we can display the autocompletes to the user. 
Does it really make sense to remove the parsing now only to add it in two weeks?

And the constant alert logs in console are the problem, then we could do either:
 - pass the flag `buildSearchQueryJSON` to fail silently and not log, `false` in every other case but this one
 - make `buildSearchQueryJSON` throw and manually handle parser errors in every place; in here we would simply do nothing

Reply:
I think we'll end up using a different parser for this that handles just autocomplete. We only really need the query AST when submitting the search to the server right?